### ISSUE TITLE
Works for main process, but not renderer, or forked processes.

### ISSUE THREAD

**Issue by petertorelli** [timfish/forge-externals-plugin#8]
Hello,
I found out about this package via this issue from the ElectronForge page that I filed:

https://github.com/electron-userland/electron-forge/issues/2949

If you read down a bit, your plug-in fixes the case when I package a native dependency invoked from the main process. However, if I move the code to the `renderer.ts` process, it cannot find `serialport` (warning from the browser); nor if I import serialport from a forked process from main, it also cannot find it (no native module exists...).

I only filed an issue with a simple bug because my main application is several years old and has grown quite a bit, but I can include portions if it would help you debug?

E.g., my `webpack.main.config.js` contains this:

```js

module.exports = {
  /**
   * This is the main entry point for your application, it's the first file
   * that runs in the main process.
   */
  entry: {
    'index': './src/main/index.ts',
    'worker-thread': './src/framework/worker-thread.ts',
    'worker-thread-lpm01a': './src/framework/device-manager/lpm01a/worker-thread.ts'
  },
  output:{
    filename: '[name].js',
  },
  module: {
    rules: require('./webpack.rules'),
  },
  // Need this so that we can build "worker-thread" which has Vue/Pug in it
  plugins: plugins,
  resolve: {
    extensions: ['.js', '.ts', '.jsx', '.tsx', '.css', '.json'],
  },
  externals: {
    serialport: "serialport",
  }
};
```

And as you can probably guess, `worker-thread-lpm01a` is forked from `worker_thread`, which is forked from `index` (main) and then imports `serialport`.

**Comment by partharanjan** [comment]
@petertorelli any luck on this issue?

**Comment by petertorelli** [comment]
@partharanjan - No, and I had forgotten about it. Thanks for the reminder, I'll close it. I closed the other issue because I migrated everything to the latest electron-vite boilerplate and started using the preload to export functions from native modules, keeping native modules out of the renderer. It is a lot of extra overhead but Electron is forcing this pattern, which means trying to go against the preferred method will be painful. Hopefully this solution actually lasts longer than a few years, but the rate at which these things change it might be smarter to switch to Qt, rather than rely buggy, ignored, and forgotten-about open source.
